common/buildcraft/factory/tile/TileChute.java

```diff
@@ -0,0 +1,212 @@
+                }
+                break;
+            }
+        }
+        return itemStack;
+    }
+
+    private void putInInventory(EnumFacing side, IInventory inventory) {
+        for(int i = 0; i < 4; i++) {
+            if(inv.getStackInSlot(i) != null) {
+                ItemStack itemStack = inv.getStackInSlot(i).copy();
+                itemStack = putStackInInventory(side, inventory, itemStack);
+                inv.setStackInSlot(i, itemStack);
+            }
+        }
+    }
+
+    private void putInInventories(EnumFacing currentSide) {
+        List<EnumFacing> sides = Arrays.asList(EnumFacing.values());
+        Collections.shuffle(sides, new Random());
+
+        for(EnumFacing side : sides) {
+            if(side == currentSide) {
+                return;
+            }
+
+            IInventory inventory = getInventoryAtPosition(worldObj, pos.offset(side));
+
+            if(inventory == null) {
+                return;
+            }
+
+            if(isInventoryFull(inventory, side)) {
+                return;
+            }
+
+            putInInventory(side, inventory);
+        }
+    }
+
+    @Override
+    public void readFromNBT(NBTTagCompound nbt) {
+        super.readFromNBT(nbt);
+        progress = nbt.getInteger("progress");
+        battery.deserializeNBT(nbt.getCompoundTag("mj_battery"));
+    }
+
+    @Override
+    public NBTTagCompound writeToNBT(NBTTagCompound nbt) {
+        super.writeToNBT(nbt);
+        nbt.setInteger("progress", progress);
+        nbt.setTag("mj_battery", battery.serializeNBT());
+        return nbt;
+    }
+
+    // ITickable
+
+    @Override
+    public void update() {
+        if(worldObj.isRemote) {
+            return;
+        }
+
+        battery.tick(getWorld(), getPos());
+
+        // test with the output of a stone engine
+        battery.addPower(1000); // remove this
+
+        EnumFacing currentSide = (EnumFacing) worldObj.getBlockState(pos).getProperties().get(BlockBCBase_Neptune.BLOCK_FACING_6);
+
+        int target = 100000;
+        if(currentSide == EnumFacing.UP.getOpposite()) {
+            progress += 1000; // can be free because of gravity
+        }
+        progress += battery.extractPower(0, target - progress);
+
+        if (progress >= target) {
+            progress = 0;
+            int radius = 3;
+            BlockPos offset = new BlockPos(currentSide.getDirectionVec());
+            offset = new BlockPos(offset.getX() * radius, offset.getY() * radius, offset.getZ() * radius);
+            AxisAlignedBB aabb = new AxisAlignedBB(this.pos, this.pos).expandXyz(radius).offset(offset);
+            List<EntityItem> entityItems = worldObj.getEntitiesWithinAABB(EntityItem.class, aabb);
+            int index = 0, max = 3;
+            for(EntityItem entityItem : entityItems) {
+                ItemStack stack = entityItem.getEntityItem();
+                stack = ItemTransactorHelper.getTransactor(this, currentSide).insert(stack, false, false);
```

Comment:
You can just use ````inv``` here (instead of ````ItemTransactorHelper.getTransactor(this, currentSide)````) as ItemHandlerSimple implements IItemTransactor already.